src/spikeinterface/core/generate.py

```diff
@@ -609,6 +609,25 @@ def generate_snippets(
 
 
 ## spiketrain zone ##
+
+
+def _ensure_firing_rates(firing_rates, num_units, seed):
+    rng = np.random.default_rng(seed=seed)
+
+    if isinstance(firing_rates, tuple):
+        rng = np.random.default_rng(seed=seed)
```

Comment:
Thisi s repeated from line 615. You can also just pass the rng instead of the seed to this function and avoid another instantiation.

Reply:
oups thank you.